File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/order/OrderDetail.java
package vn.com.ltdt.Coffee_Shop.order;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.ltdt.Coffee_Shop.product.Product;
import vn.com.ltdt.Coffee_Shop.product.ProductDetail;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "t_order_detail")
public class OrderDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int quantity;
    @NotNull(message = "The price cannot be null")
    private double price;
    @NotNull(message = "The order cannot be null")
    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
    @ManyToOne
    @JoinColumn(name = "product_detail_id")
    private ProductDetail productDetail;
    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/domains/OrderDetail.kt
package com.ltdt.coffeeshop_android_native.data.domains

import com.ltdt.coffeeshop_android_native.data.remote.OrderDetailDTO

data class OrderDetail(
    val productDetail: ProductDetail,
    val product: Product,
    var quantity: Int,
    var price: Double,
)

fun OrderDetail.toOrderDetailDTO(): OrderDetailDTO {
    return OrderDetailDTO(
        productDetail = productDetail.toProductDetailDTO(),
        product = product.toProductDTO(),
        quantity = quantity,
        price = price
    )
}
